### ISSUE TITLE
Nix version 2.13 incompatibility: "The program must call nix::initNix() before calling any libstore library functions. on attic push"

### ISSUE THREAD

**Issue by gador** [zhaofengli/attic#22]
I am getting the error

```
The program must call nix::initNix() before calling any libstore library functions.
 (core dumped) attic push remote $WORKSPACE/$val
```
(where `$WORKSPACE/$val` is a symlink to a store path which has worked just fine).

I am on revision c77b5fb64ee0d02299eee3268da2ce56b6575f6c
If there is anything to help you debug it, let me know.

The output of `attic cache info`:
```
/nix/store/m7fmj73cz1zhyahwhw6s8qs8l1gqllji-attic-0.1.0/bin/attic cache info remote
Public: false
           Public Key: remote:xxx
Binary Cache Endpoint: https://my-server.com/remote
         API Endpoint: https://my-server.com
      Store Directory: /nix/store
             Priority: 41
  Upstream Cache Keys: ["cache.nixos.org-1"]
     Retention Period: Global Default
```

**Comment by gador** [comment]
ok, this is wired:

It fails to push with the above error with:
```
> nix-shell https://github.com/zhaofengli/attic/tarball/main -A attic-client
[nix-shell:~]$ sudo -u ci attic push remote /var/lib/laminar/run/nixconfig-update/workspace/laptop
[sudo] Passwort für wogan:
The program must call nix::initNix() before calling any libstore library functions.
Aborted
```

but works with:
```
> nix-shell https://github.com/zhaofengli/attic/tarball/main -A demo

🚀 Run `atticd` to get started!


[nix-shell:~]$ sudo -u ci attic push remote /var/lib/laminar/run/nixconfig-update/workspace/laptop
✅ All done! (964 already cached, 2573 in upstream)
```

Do you have any idea? It seems the attic package from `demo` is build differently..
The store paths are:
demo: `/nix/store/hf42x6nx6bc9mz7ckn9ii5mrszvyb4rx-attic-0.1.0/bin/attic` 
attic-client: `/nix/store/1vvsrw0h0y7p1c1bnflfhnpmbh1bf0nb-attic-0.1.0/bin/attic`


Also note, that when using
```
nix build github:zhaofengli/attic
```
I get the store path /nix/store/hf42x6nx6bc9mz7ckn9ii5mrszvyb4rx-attic-0.1.0 which does not have the error message

**Comment by gador** [comment]
Ok. I guess I found culprit:

Nix recent upgrade to 2.13 broke `attic`. When build with the current nixpkgs unstable (by cloning this repo, running nix flake update and building), attic fails. 
When building with the original lock file, it passes.

When looking at the `nix-store -q --tree result` output, you can see nix version 2.12 vs. 2.13 (and other diffs, which could have potentially break attic)

**Comment by Shawn8901** [comment]
https://github.com/NixOS/nix/commit/aba6eb348e0ed8177da1e7f1df46ba00d20eab96 that seems to be the commit "causing" that, but as far as i can see, its commit is just adding a gate keeper, that programs using libstore do their initialization work correctly. 
I ran into the same issue, but somehow i dont get it fixed by removing the input override of nixpkgs override in my flake. :( so i am a bit stuck there. Also i tried to set my nix version to pre 2.13 which does also not help.

**Comment by gador** [comment]
Try adding an override and an older nixpkgs separate input. I guess by importing the Nixos module, it adds an overlay and will override an `attic` program.
I added `nixpkgs_old` as an input, set it to the revision of the lock file of attic (because it is known to work) and and the input follows of attic to nixpkgs_old.
When I added an override with something like `atitc-push = attic.programs.x86_64-linux.attic` it worked.
The `attic-push` override builds with the old nixpkgs version, while `attic` builds with the system nixpkgs.
(Sorry, I'm on mobile, but this is the gist of my workaround)

**Comment by zhaofengli** [comment]
This should be fixed in 4ad6ce9309e15021d0b5163e6939f782d6aad43e. There are some issues with building interactively with newer nixpkgs so I'm pinning the nixpkgs revision in `flake.nix` to a specific commit while I figure that out. It doesn't seem to affect sandboxed builds so overriding the input locally shouldn't be an issue.
